foundry/foundry.py

```diff
@@ -140,210 +140,126 @@ def __init__(
             force_login=False,
         )
 
-        self.xtract_tokens = {
-            "auth_token": self.auths["petrel"].access_token,
-            "transfer_token": self.auths["transfer"].authorizer.access_token,
-            "funcx_token": self.auths[
-                "https://auth.globus.org/scopes/facd7ccc-c5f4-42aa-916b-a0e270e2c2a9/all"
-            ].access_token,
-        }
+    def search(self, query: str = None, limit: int = None) -> [FoundryDataset]:
+        """Search available Foundry datasets
 
-    def load(self, name, download=True, globus=True, verbose=False, metadata=None, authorizers=None, **kwargs):
-        """Load the metadata for a Foundry dataset into the client
         Args:
-            name (str): Name of the foundry dataset
-            download (bool): If True, download the data associated with the package (default is True)
-            globus (bool): If True, download using Globus, otherwise https
-            verbose (bool): If True print additional debug information
-            metadata (dict): **For debug purposes.** A search result analog to prepopulate metadata.
+            query (str): query string to match
+            limit (int): maximum number of results to return
 
-        Keyword Args:
-            interval (int): How often to poll Globus to check if transfers are complete
-
-        Returns
-        -------
-            self
+        Returns:
+            List[FoundryDataset]: List of search results as FoundryDatset objects
         """
 
-        # handle empty dataset name (was returning all the datasets)
-        if not name:
-            raise ValueError("load: No dataset name is given")
-
-        if metadata:
-            res = metadata
-
-        # MDF specific logic
-        if is_doi(name) and not metadata:
-            res = self.forge_client.match_resource_types("dataset")
-            res = res.match_dois(name).search()
-
+        if (query is not None) and (is_doi(query)):
+            metadatas = [self.get_metadata_by_doi(query)]
```

Comment:
Do you have any additional information as to what the what the potential bug is, or how it could be addressed? Or is this just a heads up for another issue incoming?

Reply:
oh more just a note based on our previous convo, for posterity